src/main/java/uk/gov/hmcts/divorce/citizen/event/CitizenSubmitAos.java

```diff
@@ -0,0 +1,97 @@
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.CASE_WORKER;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.LEGAL_ADVISOR;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.SUPER_USER;
+import static uk.gov.hmcts.divorce.divorcecase.model.access.Permissions.CREATE_READ_UPDATE;
+import static uk.gov.hmcts.divorce.divorcecase.model.access.Permissions.READ;
+
+@Slf4j
+@Component
+public class CitizenSubmitAos implements CCDConfig<CaseData, State, UserRole> {
+
+    public static final String CITIZEN_SUBMIT_AOS = "citizen-submit-aos";
+
+    @Override
+    public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+
+        configBuilder
+            .event(CITIZEN_SUBMIT_AOS)
+            .forState(AosDrafted)
+            .name("Respondent Statement of Truth")
+            .description("The respondent confirms SOT")
+            .retries(120, 120)
+            .grant(CREATE_READ_UPDATE, APPLICANT_2)
+            .grant(READ, CASE_WORKER, SUPER_USER, LEGAL_ADVISOR)
+            .aboutToSubmitCallback(this::aboutToSubmit);
+    }
+
+    public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(CaseDetails<CaseData, State> details,
+                                                                       CaseDetails<CaseData, State> beforeDetails) {
+        log.info("Submit AOS application about to submit callback invoked");
+
+        CaseData data = details.getData();
+        AcknowledgementOfService acknowledgementOfService = data.getAcknowledgementOfService();
+        State state = details.getState();
+
+        log.info("Validating case data");
+        final List<String> validationErrors = validateAos(acknowledgementOfService);
+
+        if (!validationErrors.isEmpty()) {
+            log.info("Validation errors: ");
+            for (String error : validationErrors) {
+                log.info(error);
+            }
+
+            return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+                .data(data)
+                .errors(validationErrors)
+                .state(state)
+                .build();
+        }
+
+        state = data.getAcknowledgementOfService().getDisputeApplication() == YesOrNo.YES ? PendingDispute : Holding;
+
+        return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+            .data(data)
+            .state(state)
+            .build();
+    }
+
+    private List<String> validateAos(final AcknowledgementOfService acknowledgementOfService) {
+
+        final List<String> errors = new ArrayList<>();
+
+        if (NO.equals(acknowledgementOfService.getStatementOfTruth())) {
+            errors.add("You must be authorised by the respondent to sign this statement.");
+        }
+
+        if (NO.equals(acknowledgementOfService.getConfirmReadPetition())) {
+            errors.add("The respondent must have read the application for divorce.");
+        }
+
+        return errors;
+    }
```

Comment:
may be worth moving this to ValidationUtil class

Reply:
its kinda specific to this class hence i left it here